### PR TITLE
Fix pr comments that weren't submitted before merge

### LINKED ISSUE #12637: Review comments on 12562 and 11877
#### Problem
PR's were merged without the comments beeing submitted
* Comments for #12562
* Comments for #11877
* Replace `emberAfCurrentEndpoint` with the endpoint found in the `commandPath` for the OnOff and Color Control cluster
* Decode `commandData` inside the backend function for the color control cluster and On Off cluster instead of in the callback function

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* Fixes #12637 

#### Change overview
* Removed `emberAfCurrentEndpoint` in OnOff and ColorControl
* Decode commandData inside the backend function

#### Testing
* No fonctionnal changes
* Automated tests to validate no regressions were caused

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 1]
```diff
@@ -2408,158 +2457,97 @@ void emberAfPluginColorControlServerStopTransition(void)
...
 {
-    auto & moveMode        = commandData.moveMode;
-    auto & rate            = commandData.rate;
-    auto & optionsMask     = commandData.optionsMask;
-    auto & optionsOverride = commandData.optionsOverride;
-
-    return ColorControlServer::Instance().moveHueCommand(moveMode, static_cast<uint16_t>(rate), optionsMask, optionsOverride,
-                                                         false);
+    return ColorControlServer::Instance().moveHueCommand(commandPath.mEndpointId, commandData.moveMode,
+                                                         static_cast<uint16_t>(commandData.rate), commandData.optionsMask,
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:2461]: Followup to remove that cast?

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 2]
```diff
@@ -2408,158 +2457,97 @@ void emberAfPluginColorControlServerStopTransition(void)
...
 {
-    auto & hue             = commandData.hue;
-    auto & hueMoveMode     = commandData.direction;
-    auto & transitionTime  = commandData.transitionTime;
-    auto & optionsMask     = commandData.optionsMask;
-    auto & optionsOverride = commandData.optionsOverride;
-
-    return ColorControlServer::Instance().moveToHueCommand(static_cast<uint16_t>(hue), hueMoveMode, transitionTime, optionsMask,
-                                                           optionsOverride, false);
+    return ColorControlServer::Instance().moveToHueCommand(commandPath.mEndpointId, static_cast<uint16_t>(commandData.hue),
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:2476]: Followup to remove that cast?

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 3]
```diff
@@ -2408,158 +2457,97 @@ void emberAfPluginColorControlServerStopTransition(void)
...
-    auto & hue             = commandData.hue;
-    auto & saturation      = commandData.saturation;
-    auto & transitionTime  = commandData.transitionTime;
-    auto & optionsMask     = commandData.optionsMask;
-    auto & optionsOverride = commandData.optionsOverride;
-
-    return ColorControlServer::Instance().moveToHueAndSaturationCommand(static_cast<uint16_t>(hue), saturation, transitionTime,
-                                                                        optionsMask, optionsOverride, false);
+    return ColorControlServer::Instance().moveToHueAndSaturationCommand(
+        commandPath.mEndpointId, static_cast<uint16_t>(commandData.hue), commandData.saturation, commandData.transitionTime,
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:2493]: Followup to remove that cast.

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 4]
```diff
@@ -2408,158 +2457,97 @@ void emberAfPluginColorControlServerStopTransition(void)
...
-    auto & stepMode        = commandData.stepMode;
-    auto & stepSize        = commandData.stepSize;
-    auto & transitionTime  = commandData.transitionTime;
-    auto & optionsMask     = commandData.optionsMask;
-    auto & optionsOverride = commandData.optionsOverride;
-
     return ColorControlServer::Instance().stepHueCommand(
-        stepMode, static_cast<uint16_t>(stepSize), static_cast<uint16_t>(transitionTime), optionsMask, optionsOverride, false);
+        commandPath.mEndpointId, commandData.stepMode, static_cast<uint16_t>(commandData.stepSize),
+        static_cast<uint16_t>(commandData.transitionTime), commandData.optionsMask, commandData.optionsOverride, false);
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:2502]: Followup to remove the casts.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 1 above)
Comment (mkardous-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp]: done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 2 above)
Comment (mkardous-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp]: done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 3 above)
Comment (mkardous-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp]: done

File: src/app/clusters/color-control-server/color-control-server.cpp (hunk 4 above)
Comment (mkardous-silabs) [reply src/app/clusters/color-control-server/color-control-server.cpp]: done

**Review by rcasallas-silabs — APPROVED** [review]

**Pull request merged**